File: Stack/Stack.cs
namespace Struct.Stack;

class Stack
{
    private readonly int _maxSize;
    private int _top;
    private readonly int[] _stackArray;
    
    /*
     Initializes a new instance of the Stack class with the specified size.
    Parameters:
    size:
    The maximum size of the stack.
    */ 

    public Stack(int size)
    {
        _maxSize = size;
        _top = -1;
        _stackArray = new int[_maxSize];
    }

    // Pushes an integer value onto the stack.
    public void Push(int value)
    {
        if (_top == _maxSize - 1)
        {
            Console.WriteLine("Стек полон. Невозможно добавить элемент.");
            return;
        }

        _stackArray[++_top] = value;
    }
    /*
    Pop function pops the top element from the stack and returns it.
    If the stack is empty, it prints an error message and returns -1.
    Returns:
         The top element of the stack if the stack is not empty.
         -1 if the stack is empty.
    */
    public int Pop()
    {
        if (_top == -1)
        {
            Console.WriteLine("Стек пуст. Невозможно извлечь элемент.");
            return -1;
        }

        return _stackArray[_top--];
    }
    /*
    Retrieves the top element from the stack without removing it.
    Returns:
          The top element of the stack.
          If the stack is empty, returns -1 and prints an error message.
     */
    public int Peek()
    {
        if (_top == -1)
        {
            Console.WriteLine("Стек пуст. Невозможно получить элемент.");
            return -1;
        }

        return _stackArray[_top];
    }

    public bool IsEmpty()
    {
        return (_top == -1);
    }

    public bool IsFull()
    {
        return (_top == _maxSize - 1);
    }
}

File: Stack/Stack.py
class Stack:
    """
    Initializes a new instance of the class.
    Parameters:
            self: The object itself.
    Returns:
            None
    """
    def __init__(self):
        self.stack = []
    """
    Check if the stack is empty.
    Returns:
            bool: True if the stack is empty, False otherwise.
    """
    def is_empty(self):
        return len(self.stack) == 0
    """
    Pushes an item onto the stack.
    Parameters:
            item (any): The item to be pushed onto the stack.
    Returns:
            None
    """

    def push(self, item):
        self.stack.append(item)
    """
    Removes and returns the top element of the stack.
    Parameters:
            None
    Returns:
            The top element of the stack, or None if the stack is empty.
    """
    def pop(self):
        if self.is_empty():
            return None
        return self.stack.pop()
    """
    Returns the top element of the stack without removing it.
    Returns:
            The top element of the stack, or None if the stack is empty.
    """

    def peek(self):
        if self.is_empty():
            return None
        return self.stack[-1]
    """
    Returns the size of the stack.
    Returns:
            int: The size of the stack.
    """
    def size(self):
        return len(self.stack)